src/ring_theory/ideal/operations.lean

```diff
@@ -845,6 +845,16 @@ lemma map_map {T : Type*} [ring T] {I : ideal R} (f : R →+* S)
 ((gc_map_comap f).compose _ _ _ _ (gc_map_comap g)).l_unique
   (gc_map_comap (g.comp f)) (λ _, comap_comap _ _)
 
+@[simp] lemma map_span (f : R →+* S) (s : set R) :
+  map f (span s) = span (f '' s) :=
+symm $ submodule.span_eq_of_le _
+  (λ y ⟨x, hy, x_eq⟩, x_eq ▸ mem_map_of_mem f (subset_span hy))
+  (map_le_iff_le_comap.2 $ span_le.2 $ set.image_subset_iff.1 subset_span)
+
+@[simp] lemma map_span' (f : R →+* S) (s : set R) :
+  map f (submodule.span R s) = submodule.span S (f '' s) :=
```

Comment:
Shouldn't we have a simp-lemma that simps `submodule.span` into `ideal.span`, in situations like this?